pkg/virt-controller/network/ipamclaims.go

```diff
@@ -0,0 +1,206 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+
+	"kubevirt.io/client-go/kubecli"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	ipamclaims "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1"
+)
+
+type NetConf struct {
+	AllowPersistentIPs bool `json:"allowPersistentIPs,omitempty"`
+}
+
+func GetNetworkToIPAMClaimName(virtClient kubecli.KubevirtClient, namespace string, name string, networks []virtv1.Network, persistentIPsEnabled bool) (map[string]string, error) {
+	networkToIPAMClaimName := make(map[string]string)
+	for _, network := range networks {
+		if vmispec.IsSecondaryMultusNetwork(network) {
+			allowPersistentIPs, err := getAllowPersistentIPs(virtClient, namespace, network.Multus.NetworkName)
+			if err != nil {
+				return map[string]string{}, fmt.Errorf("GetNetworkToIPAMClaimName failed: %w", err)
```

Comment:
This string should be set at the caller not at the callee, since this is what you are doing here https://github.com/kubevirt/kubevirt/pull/11410/commits/e73fcdb46eab77980dca5ca8e72c26d0454a6bab#diff-16487fff595728eb5eeec01225a2203756ecad25d90e90d5feb13a9ff08df7ffR2374

Also in general in golang errors start with "failed ..."

So this should be 
```
failed retrieving allowPersistentIPs flag
```